### DISCUSSION TITLE
Better checks support for Bitbucket platform in Dependency Dashboard

### DISCUSSION THREAD

**Discussion by ArtyomGabeev** [renovatebot/renovate#29086] [Suggest an Idea]
### Tell us more.

I'm using renovate bot on Bitbucket platform.
When dependency dashboard is created, I see pending PRs can be forced to be processed by checking the checkbox, e.g:
```
Pending Status Checks
These updates await pending status checks. To force their creation now, click the checkbox below.
RENOVATE Update dependency scala to v3.4.2
```
Unfortunately, Bitbucket does not support checkboxes extension for markdown (and I think will never be):
https://jira.atlassian.com/browse/BCLOUD-8586

The only way I see we can support checks controls is via issue comments. 
Looking for ideas/suggestions.

**Comment by rarkins** [comment]
Bitbucket Cloud, or Bitbucket Data Center?

**Reply by ArtyomGabeev** [reply]
Bitbucket Cloud

**Reply by rarkins** [reply]
For Bitbucket Cloud we have built the equivalent to the Dependency Dashboard into the (free) Mend.io hosted app: https://marketplace.atlassian.com/apps/1232072/mend?hosting=cloud&tab=overview

Docs: https://docs.mend.io/bundle/platform/page/install_mend_developer_platform_for_bitbucket_cloud.html

i.e. you can open PRs on-demand through the web UI at developer.mend.io

**Reply by ArtyomGabeev** [reply]
This option won't work with self-hosted runners, right?

**Reply by rarkins** [reply]
No, we have built this as part of our multi-tenant SaaS solution. Renovate use on this platform is completely free though.

**Reply by ArtyomGabeev** [reply]
I appreciate team effort, but I have a hard requirement to run self-hosted renovate.
Can we update check regexp to include "!" or "!open" in additional to checkbox?
Also we may update wordings:
"These updates await pending status checks. To force their creation now, click the checkbox below, or add "!open" if your platform does not support checkboxes."

It looks to me as simple change, which will work for most of non github users.

Thanks.

**Reply by rarkins** [reply]
Do you mean you'd add an issue comment to the dashboard issue in order to open something on the dashboard?

Would these comments continue forever? (maybe into hundreds)

How would you identify a particular PR? e.g. by copy pasting the PR title?

Because PR titles can cycle, would Renovate ignore all but the latest comment? Latest 5 comments?

**Reply by ArtyomGabeev** [reply]
Looking at depependency dashboard at this repo:
https://github.com/renovatebot/renovate/issues/2958

What I suggest is to allow to use "!" or "!open" before PR title. Assumption that user should edit issue description manually.
E.g.:
```
Pending Status Checks
These updates await pending status checks. To force their creation now, click the checkbox below or add "!open" before the branch name.

- !open build(deps): update dependency redis to v4.6.14
```

**Reply by rarkins** [reply]
One challenge is that PR titles aren't always unique and could in theory occur more than once on a Dashboard. Uniqueness is instead based on branch name, so if you look behind the scenes in the GitHub dashboard you see it like this: `- [ ] <!-- approve-branch=renovate/chalk-5.x -->`. It's probably not a blocker though - the matching wouldn't be as exact but I think it would be correct in 99% of cases.

**Comment by HonkingGoose** [comment]
Related issues or discussions:

- #1355
- #20166
- #20376
